packages/aic/src/document.ts

```diff
@@ -57,50 +57,51 @@ makes
 
 export const updateTick = async (
   did: string, // the did:aic: being updated
-  tid: TidString, // the consenus time of the tick's generation
+  tid: TidString, // the consensus time of the tick's generation
   candidateDiff: Diff | Document | null, // null when only updating tid
-  stored_tick: Tick | null, // null when there is not db entry (did init)
+  storedTick: Tick | null, // null when there is not db entry (did init)
   key: Asymmetric, // the consortium passes in the key
-): Promise<Tick | ErrorMessage> => {
+): Promise<Tick> => {
   // @TODO validate that tid > all tids in the diffs
-  if (stored_tick === null) {
+  if (storedTick === null) {
     // this is the  initial registration of a did:aic
     return registerNewDid(did, tid, candidateDiff, key)
   }
 
-  // since stored_tick is not null this is an update
+  // since storedTick is not null this is an update
   // of a registered did:aic
-  if (!(await validateSig(stored_tick, key))) {
-    // this means the consortium data base has been corupted
-    return { error: 'bad signature: stored tick' }
+  if (!(await validateSig(storedTick, key))) {
+    // this means the consortium database has been corrupted
+    throw new Error('bad signature: stored tick')
   }
-  if (stored_tick.key !== key.did()) {
-    // the consortium should not update a tick signed by a difrent consortium
+  if (storedTick.key !== key.did()) {
+    // the consortium should not update a tick signed by a different consortium
     // this logic may get more complicated if the consortium is mid key rotation
-    return { error: 'mismatch: consortium key, stored tick key' }
+    throw new Error('mismatch: consortium key, stored tick key')
   }
-  if (stored_tick.did !== did) {
-    // databace index is coruptend
-    return {
-      error: 'stored tick not for did',
+  if (storedTick.did !== did) {
+    // database index is corrupted
+    const error = new Error('stored tick not for did')
+    Object.assign(error, {
       did: did,
-      stored_tick_did: stored_tick.did,
-    }
+      storedTickDid: storedTick.did,
+    })
+    throw error
   }
-  // stored_tick is now validated by consortium key and url
-  return updateExistingValidatedDid(tid, candidateDiff, stored_tick, key)
+  // storedTick is now validated by consortium key and url
+  return updateExistingValidatedDid(tid, candidateDiff, storedTick, key)
 }
 
 const registerNewDid = async (
   did: string,
   tid: TidString,
   candidateDiff: Document | null,
   key: Asymmetric,
-): Promise<Tick | ErrorMessage> => {
+): Promise<Tick> => {
   if (candidateDiff === null) {
     // since candidateDiff is null
-    // this is a request for a fresh tick of a non-exixtant did
-    // return a signed affirmation that the did dose not exist
+    // this is a request for a fresh tick of a non-existent did
+    // return a signed affirmation that the did does not exist
```

Comment:
Since we changed the errors to throws this is not an error

We should change the API from 
```
{
        tid: tid,
        did: did,
        error: 'not found',
        key: key.did(),
        sig: '',
      },
```
To
```
{
        tid: tid,
        did: did,
        diffs: {},
        key: key.did(),
        sig: '',
      },
```
It's not an error it's a affirmative claim that there are no stored diffs even the origin diff